generics/services/questionset.js

```diff
@@ -0,0 +1,175 @@
+        "content-type": "application/json",
+        "Authorization": process.env.CREATION_PORTAL_AUTHORIZATION_KEY
+      }
+      function updateQuestionSetHierarchyCallBack(err, data) {
+        if (err || data.statusCode != 200) {
+          return reject({
+            message: messageConstants.apiResponses.QUESTIONSET_NOT_FOUND,
+            status: httpStatusCode.bad_request.status,
+          })
+        } else if (data.statusCode == 200) {
+          return resolve({
+            status: data.statusCode
+          })
+        }
+      }
+      request.patch(updateUrl, { headers: headers, json: true, json: req }, updateQuestionSetHierarchyCallBack)
+
+    } catch (error) {
+      return reject(error);
+    }
+  })
+
+}
+
+const publishQuestionSet = function (questionsetId) {
+  return new Promise((resolve, reject) => {
+    try {
+
+      let publishUrl = `${CREATION_PORTAL_URL}${messageConstants.endpoints.PUBLISH_QUESTION_SET}/${questionsetId}`;
+      const headers = {
+        "content-type": "application/json",
+        "Authorization": process.env.CREATION_PORTAL_AUTHORIZATION_KEY
+      }
+      async function publishQuestionSetCallBack(err, data) {
+        if (err || data.statusCode != 200) {
+          const errmsg = JSON.parse(data.body)
+          return reject({
+            message: errmsg.errmsg || messageConstants.apiResponses.QUESTIONSET_NOT_FOUND,
+            status: httpStatusCode.bad_request.status,
+          })
+        } else if (data.statusCode == 200) {
+          return resolve({
+            status: data.statusCode
+
+          })
+        }
+      }
+      request.post(publishUrl, { headers: headers }, publishQuestionSetCallBack)
+
+    } catch (error) {
+      return reject(error);
+    }
+  })
+
+}
+
+const updateQuestionSet = function (updateReq, migratedId) {
+  return new Promise((resolve, reject) => {
+    try {
+      const headers = {
+        "content-type": "application/json",
+        "Authorization": process.env.CREATION_PORTAL_AUTHORIZATION_KEY
+      }
+      let updateUrl = CREATION_PORTAL_URL + messageConstants.endpoints.UPDATE_QUESTION_SET + "/" + migratedId;
+      const options = {
+        headers,
+        json: true,
+        json: { request: { questionset: updateReq } }
+      };
+      function updateQuestionSetCallBack(err, data) {
+        if (err || data.statusCode != 200) {
+          return reject({
+            message: messageConstants.apiResponses.QUESTIONSET_NOT_FOUND,
+            status: httpStatusCode.bad_request.status,
+          })
+        } else if(data.statusCode == 200) {
+            return resolve({
+              status: data.statusCode
+            })
+        }
+      }
+      request.patch(updateUrl, options, updateQuestionSetCallBack)
+
+    } catch (error) {
+      return reject(error);
+    }
+  })
+}
+
+module.exports = {
+  copyQuestionSet: copyQuestionSet,
+  readQuestionSet: readQuestionSet,
+  updateQuestionSetHierarchy: updateQuestionSetHierarchy,
```

Comment:
@swetha-aggidevara for what use case we need updateQuestionSetHierarchy ?

Reply:
This is Aishwarya's code usage, we are using the same file